x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/predicate/operator/arithmetic/Arithmetics.java

```diff
@@ -162,4 +152,26 @@ static Number negate(Number n) {
 
         return Integer.valueOf(Math.negateExact(n.intValue()));
     }
+
+    private static Temporal periodArithmetics(Temporal l, Period r, BiFunction<Temporal, Period, Temporal> f) {
+        if (l == null || r == null) {
+            return null;
+        }
+
+        if (l instanceof OffsetTime) {
+            r = r.withDays(0).withMonths(0).withYears(0);
```

Comment:
This creates  2 temporary objects - use `of` or `Instant` instead.

Reply:
This is a Period, couldn't find an alternative way.

Reply:
I was confused that Period can also have hours,mins, etc..
Thx and apologies for that.